### DISCUSSION TITLE
always get black thumbnail

### DISCUSSION THREAD

**Discussion by klydeid** [OpenImageIO/oiio#3506] [Q&A]
Hi. i want to save the thumbnail image from PSD file to disk by use ImageInput::get_thumbnail() function.
and the result is black. and when i print the pixel color always get zero too.
i'm pretty sure i have checked the thumbnail option when i save these PSD file from photoshop.

here is my simple code. what is wrong?

    using namespace OIIO;

    auto inp = ImageInput::open("C:/Users/Admin/Desktop/thumbnail_PSD.psd");
    
    ImageBuf thumb_buf = ImageBuf();
    bool getthumb = inp->get_thumbnail(thumb_buf, 0);

    if (getthumb) {
        std::cout << "Yes" << std::endl;
        for (int i = 0; i < thumb_buf.spec().height; i++) {
            for (int j = 0; j < thumb_buf.spec().width; j++) {
                std::cout << thumb_buf.getchannel(j, i, 1, 0) << std::endl;
            }
        }
    }

    else {
        std::cout << "No" << std::endl;
    }

**Comment by lgritz** [comment]
All psd files? Just one? Can you send a (small) example PSD file that exhibits this issue?

What version of OIIO?

**Reply by lgritz** [reply]
Any follow-up on this? Can you provide an example file that exhibits the problem?

**Reply by klydeid** [reply]
Hi.
I'm so sorry for the later reply.
here is a PSD file i have tested.
https://www.dropbox.com/s/tmmk3u9jbxb34sr/thumbnail_PSD.rar?dl=0

check that.
and my OIIO version is 2.3.15
